src/dll_api.cpp

```diff
@@ -69,47 +69,12 @@ WIN_SPARKLE_API void __cdecl win_sparkle_init()
         // first things first
         UpdateDownloader::CleanLeftovers();
 
-        // check for updates
-        bool checkUpdates;
-        if ( Settings::ReadConfigValue("CheckForUpdates", checkUpdates) )
-        {
-            if ( checkUpdates )
-            {
-                static const time_t ONE_DAY = 60*60*24;
-
-                time_t lastCheck = 0;
-                Settings::ReadConfigValue("LastCheckTime", lastCheck);
-                const time_t currentTime = time(NULL);
-
-                // Only check for updates in reasonable intervals:
-                const int interval = win_sparkle_get_update_check_interval();
-                if ( currentTime - lastCheck >= interval )
-                {
-                    // Run the check in background. Only show UI if updates
-                    // are available.
-                    UpdateChecker *check = new UpdateChecker();
-                    check->Start();
-                }
-            }
-        }
-        else // not yet configured
-        {
-            bool didRunOnce;
-            Settings::ReadConfigValue("DidRunOnce", didRunOnce, false);
-            if ( !didRunOnce )
-            {
-                // Do nothing on the first execution of the app, for better
-                // first-time impression.
-                Settings::WriteConfigValue("DidRunOnce", true);
-            }
-            else
-            {
-                // Only when the app is launched for the second time, ask the
-                // user for their permission to check for updates.
-                UI::AskForPermission();
-            }
-        }
+        // Run the check in background. Only show UI if updates
+        // are available.
+        UpdateChecker *check = new UpdateChecker();
```

Comment:
Hmm, this launches the updates checking thread even if the user explicitly said they don’t want that. I see the logic behind it (if the user later changes their mind, the checker thread will pick the change up), but it’s still a bit wasteful, isn’t it? 

IOW, which is the lesser evil: always needlessly creating a thread that will never be used, or re-enabling automatic checks only after the user restarts the application? I’m inclined towards the latter, but maybe I’m missing something?